go/pkg/cas/upload.go

```diff
@@ -375,21 +384,55 @@ func (c *Client) Upload(ctx context.Context, opt UploadOptions, inputC <-chan *U
 			u.batchBundler.Flush() // only after wgChecks is done.
 		}()
 
-		for {
-			select {
-			case <-ctx.Done():
-				return ctx.Err()
-			case in, ok := <-inputC:
-				if !ok {
-					return nil
+		ueg, ctx := errgroup.WithContext(ctx)
```

Comment:
I think here you want to create a `gCtx` and use it instead of `ctx` inside the goroutines you launch with `ueg.Go`.

Reply:
as discussed, using a non-cancelling `errgroup.Group` fixes it.  Uploaded.